airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StandardSyncPersistence.java

```diff
@@ -214,4 +253,40 @@ private void writeStandardSync(final StandardSync standardSync, final DSLContext
     }
   }
 
+  private Stream<StandardSyncIdsWithProtocolVersions> findDisabledSyncs(final DSLContext ctx, final UUID actorDefId, final ActorType actorType) {
+    // Table aliasing to help have a readable join
+    final Actor source = ACTOR.as("source");
+    final Actor destination = ACTOR.as("destination");
+    final ActorDefinition sourceDef = ACTOR_DEFINITION.as("sourceDef");
+    final ActorDefinition destDef = ACTOR_DEFINITION.as("destDef");
+
+    // Retrieve all the connections currently disabled due to a bad protocol version
+    // where the actor definition is matching the one provided to this function
+    final Stream<StandardSyncIdsWithProtocolVersions> results = ctx
+        .select(CONNECTION.ID, sourceDef.ID, sourceDef.PROTOCOL_VERSION, destDef.ID, destDef.PROTOCOL_VERSION)
+        .from(CONNECTION)
+        .join(source).on(CONNECTION.SOURCE_ID.eq(source.ID))
+        .join(sourceDef).on(source.ACTOR_DEFINITION_ID.eq(sourceDef.ID))
```

Comment:
Why do we need the join on the source / source def ? I think that it is missing in the where clause.